BusinessEvents.DataStream/DataStreamProcessor.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Threading.Tasks;
+using Amazon.Lambda;
+using Amazon.Lambda.DynamoDBEvents;
+using PageUp.Events;
+using BusinessEvents.SubscriptionEngine;
+using Newtonsoft.Json;
+using BusinessEvents.DeadLetter;
+using BusinessEvents.EventStore;
+
+namespace BusinessEvents.DataStream
+{
+    public class DataStreamProcessor : IDataStreamProcessor
+    {
+        private readonly ISubscriptionProcessor subscriptionProcessor;
+        private readonly IDeadLetterService deadLetterService;
+
+        public DataStreamProcessor(ISubscriptionProcessor subscriptionProcessor, IDeadLetterService deadLetterService)
+        {
+            this.subscriptionProcessor = subscriptionProcessor;
+            this.deadLetterService = deadLetterService;
+        }
+
+        private async Task MarkAsDeadLetter(Event @event, string function, Exception exception = null)
+        {
+            var message = JsonConvert.SerializeObject(@event);
+
+            var deadLetter = new DeadLetterMessage
+            {
+                Function = function,
+                MessageId = @event?.Message?.Header?.MessageId,
+                PublishedTimeStampUtc = @event?.Header.TransportTimeStamp,
+                CreatedTimeStampUtc = DateTime.UtcNow,
+                Domain = BusinessEventStore.GetDomain(@event),
+                InstanceId = @event?.Header?.InstanceId,
+                MessageType = @event?.Message?.Header?.MessageType,
+                Message = message,//.Encrypt().ToCompressedBase64String(),
+                Exception = exception
+            };
+
+            try
+            {
+                await deadLetterService.Handle(deadLetter);
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(
+                    $"Error: MessageId: {@event?.Message?.Header?.MessageId} MarkAsDeadLetter Failed: {JsonConvert.SerializeObject(e)} DeadLetter: {JsonConvert.SerializeObject(deadLetter)}");
+            }
+        }
+
+        public async Task Process(DynamoDBEvent dynamoDbEvent, string awsAccountId)
+        {
+            Environment.SetEnvironmentVariable("ACCOUNT_ID", awsAccountId);
```

Comment:
i think we can set ACCOUNT_ID in handler level so we don't have to pass in the awsAccountId into the Process method. If any processor requires account_id, we could get it from the environment collection.

Reply:
making the dependency explicit atleast ensures we don't miss it, imo.

Reply:
alternatively, I am thinking to make it a dependency of datastream processor instead of process method. what say?